mteb/abstasks/AbsTaskZeroshotClassification.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import tqdm
+from datasets import Dataset
+
+from mteb.encoder_interface import Encoder, EncoderWithQueryCorpusEncode
+from mteb.load_results.mteb_results import ScoresDict
+
+from ..evaluation.evaluators import ZeroshotClassificationEvaluator
+from .AbsTask import AbsTask
+
+logger = logging.getLogger(__name__)
+
+
+class AbsTaskZeroshotClassification(AbsTask):
+    """Abstract class for ZeroshotClassification tasks
+    The similarity between an images and candidate text prompts, such as this is a dog/this is a cat.
+
+    self.load_data() must generate a huggingface dataset with a split matching self.metadata_dict["eval_splits"], and assign it to self.dataset. It must contain the following columns:
+        png: list of Image.Image
+        cls: list of int
+    """
+
```

Comment:
```suggestion
   Attributes:
        png: A list of Image.Image (describe the format here)
        cls: list of int (denoting what)
    """
    cls: list[int]
    png: list[Image] 
```
also why not call it images and labels?

Reply:
Even though the RenderedSST2 HF dataset has `cls` and `png` column names, I agree that this is a good opportunity for us to standardize these at the MIEB level. Other datasets uses different column names, e.g. https://huggingface.co/datasets/clip-benchmark/wds_flickr8k uses `jpg` and `txt`.

Reply:
@KennethEnevoldsen @isaac-chung definetely agree. Will change it in the next push. Only reason to use this was because was working on this single task for proof of concept quickly.

I will also change `png` and `cls` in the following to general names (maybe something users can define in the dataset class in __init__ when adding their own dataset?)  
```python
        evaluator = ZeroshotClassificationEvaluator(
            dataset["png"],  
            dataset["cls"],  
            text_candidates,
            task_name=self.metadata.name,
            **kwargs,
        )
```

such as something like

```python
        evaluator = ZeroshotClassificationEvaluator(
            dataset[self.image_column_name],  
            dataset[self.label_column_name],  
            text_candidates,
            task_name=self.metadata.name,
            **kwargs,
        )
```

Reply:
That seems reasonable something like:

```
class MyZeroShotTask(ZeroShotClassificationAbstask):
   image_column_name = "image"
   label_column_name = "label"
   
   metadata = TaskMetadata(...)

Reply:
have [adjusted](https://github.com/embeddings-benchmark/mteb/blob/mieb/mteb/abstasks/Image/AbsTaskZeroshotClassification.py#L27C24-L27C27) with a default `image_column_name`, `label_column_name` which can be [overridden](https://github.com/embeddings-benchmark/mteb/blob/mieb/mteb/tasks/Image/ZeroshotClassification/eng/RenderedSST2.py#L37C28-L37C29) in the dataset class.